libs/s25main/QuickStartGame.cpp

```diff
@@ -87,8 +87,10 @@ bool QuickStartGame(const boost::filesystem::path& mapOrReplayPath, const std::v
 
     WINDOWMANAGER.Switch(std::make_unique<dskSelectMap>(csi));
 
-    if((extension == ".sav" && GAMECLIENT.HostGame(csi, mapOrReplayPath, MapType::Savegame))
-       || ((extension == ".swd" || extension == ".wld") && GAMECLIENT.HostGame(csi, mapOrReplayPath, MapType::OldMap)))
+    if((extension == ".sav"
+        && GAMECLIENT.HostGame(csi, mapOrReplayPath, boost::optional<boost::filesystem::path>(), MapType::Savegame))
+       || ((extension == ".swd" || extension == ".wld")
+           && GAMECLIENT.HostGame(csi, mapOrReplayPath, boost::optional<boost::filesystem::path>(), MapType::OldMap)))
```

Comment:
Use `boost::none` for an empty optional which is much easier to read:

```suggestion
           && GAMECLIENT.HostGame(csi, mapOrReplayPath, boost::none, MapType::OldMap)))
```